### PR TITLE
Unvendor ppx_tools_versioned

### PR DESCRIPTION
**Pull request by tekknolagi**
Shout-out to @let-def for getting that on opam!

### CODE REVIEW

**Comment by tekknolagi** [comment]
@let-def looks like the tagged release available on opam (called "5.0alpha" but actually "5.0" and lacking archive fields in the META file) is hard to use. Would it be possible to release a 5.1 on opam that has the archive(byte...) fields?

**Comment by tekknolagi** [comment]
@yunxing @jordwalke @chenglou weird issue: CI "passes" even though project failed to build. Hmm...

**Comment by tekknolagi** [comment]
And it builds! Why? I don't know! :D

**Comment by let-def** [comment]
@tekknolagi: it is not lacking archive fields. And I plan to call it 5.0 when I am more confident in the release, alpha is fine for now :).

**Comment by tekknolagi** [comment]
@let-def here is what I mean. The tree tagged 5.0alpha, which appears to have been released on opam, has a META file that does not have archive fields: https://github.com/let-def/ppx_tools_versioned/blob/5.0alpha/META. If in the opam repo there is a different version available, please let me know.

**Comment by tekknolagi** [comment]
Actually it looks like while trunk does (https://github.com/let-def/ppx_tools_versioned/blob/b7946d25ff7dc0adaf998816f2ed39f6c0002f22/META), ppx_tools does not have them and ppx_deriving does OK without them.

**Comment by let-def** [comment]
Ah yes you are misunderstanding the purpose of the fields.
I try to explain that a bit in [MANUAL](https://github.com/let-def/ocaml-migrate-parsetree/blob/master/MANUAL.md), especially [findlib conventions](https://github.com/let-def/ocaml-migrate-parsetree/blob/master/MANUAL.md#conventions-for-distributing-a-linkable-ppx-rewriter).

Shortly:
- a ppx rewriter might not need any object at all: it is rewriting the AST but it is not affecting the rest of the pipeline. `archive` fields are only for specifying objects to link
- in trunk, the `archive` fields have a different purpose: making it possible to link your own ppx rewriter. In a normal use, (without the *omp_driver* and *custom_ppx* predicates), they are ignored.

**Comment by tekknolagi** [comment]
@let-def yes but also ppx_deriving links against Ast_convenience, right?

**Comment by let-def** [comment]
Yes, but *ppx_deriving* requires *ppx_tools*/*ppx_tools_versioned* which contains the archive field.

File: package.json.in [hunk 1]
```diff
@@ -54,6 +54,7 @@
         "@opam-alpha/ocaml-migrate-parsetree": "*",
         "@opam-alpha/result": "*",
         "@opam-alpha/topkg": "0.8.1",
+        "@opam-alpha/ppx_tools_versioned": "5.0alpha",
```
Comment (yunxing) [inline package.json.in]: Please use 5.0.0 here. 5.0 is not a valid semantic version and we converted the package as version "5.0.0": https://github.com/npm-opam/ppx_tools_versioned/blob/latest/package.json

**Comment by tekknolagi** [comment]
@let-def I don't see the archive field in ppx_tools_versioned, though. Can you point me to it?

File: package.json.in (hunk 1 above)
Comment (tekknolagi) [inline package.json.in]: Thanks for pointing this out. Will fix.

**Comment by let-def** [comment]
@tekknolagi https://github.com/let-def/ppx_tools_versioned/blob/b7946d25ff7dc0adaf998816f2ed39f6c0002f22/META#L3-L4

**Comment by tekknolagi** [comment]
Is the blob tagged 5.0alpha in ppx_tools_versioned the blob that is in opam-repository?

**Comment by yunxing** [comment]
Tested this on npm and opam on my machine, works well.

**Comment by let-def** [comment]
@tekknolagi Yes (you can check with `opam source ppx_tools_versioned`). Are you looking for something specific?!

**Comment by tekknolagi** [comment]
@let-def Yeah I am trying to resolve my confusion. opam and friends have been a bit of a struggle for me. I think, though, it turns out I am just blind and there are noarchive fields in the package bit but they are instead at the top of the META file. I am sorry for wasting your time here! I clearly cannot read 🙃

**Comment by let-def** [comment]
You are welcome :). I hope it cleared the confusion, don't hesitate to ask questions.

**Comment by yunxing** [comment]
Thanks for cleaning this up!

**Pull request merged**